### PR TITLE
[SD] separate concerns in math/statistics/lists packages.

### PR DESCRIPTION
**Pull request by SDooman**
Some of the numeric helper functions are more generic than "statistical" functions.  Therefore (for the time being), we should label this package as `math`, and down the ride if the need for a specific `statistics` package comes up, the two can diverge as such.

### CODE REVIEW

**Comment by jpolitz** [comment]
Can you also rename it in the documentation @dooman?  @shriram – you may need to change your link after all, depending on how we name things and what your stats link was referring to.

**Comment by shriram** [comment]
Yes. Please let me know when the new docs are actually live, and I'll
update my link. Thanks.

File: tests/pyret/tests/test-statistics.arr [hunk 1]
```diff
@@ -1,53 +1,36 @@
 import equality as E
```
Comment (sorawee) [inline tests/pyret/tests/test-statistics.arr]: This whole file need to be rewritten.

File: tests/pyret/main2.arr [hunk 2]
```diff
@@ -18,7 +18,7 @@ import file("./tests/test-equality.arr") as _
 import file("./tests/test-refined-refs.arr") as _
 import file("./tests/test-match.arr") as _
 import file("./tests/test-lists.arr") as _
-import file("./tests/test-statistics.arr") as _
```
Comment (sorawee) [inline tests/pyret/main2.arr]: `import file("./tests/test-statistics.arr") as _` should still be there.

File: tests/pyret/main.arr [hunk 3]
```diff
@@ -10,7 +10,7 @@ import "./tests/test-output.arr" as _
 import "./tests/test-array.arr" as _
 import "./tests/test-constructors.arr" as _
 import "./tests/test-contracts.arr" as _
-import "./tests/test-statistics.arr" as _
```
Comment (sorawee) [inline tests/pyret/main.arr]: import file("./tests/test-statistics.arr") as _

File: src/arr/trove/lists.arr [hunk 4]
```diff
@@ -812,6 +812,19 @@ fun filter-values<a>(lst :: List<Option<a>>) -> List<a>:
   end
 end  
 
+fun distinct(l :: List) -> List:
```
Comment (sorawee) [inline src/arr/trove/lists.arr]: Would be nice to use parametric polymorphism:

```
fun distinct<A>(l :: List<A>) -> List<A>:
   ...
end
```

File: tests/pyret/tests/test-math.arr [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  math.max([list:]) raises "empty"
+  math.min([list:]) raises "empty"
+
+  math.max([list: 1, 2, ~3]) is-roughly ~3
+  math.max([list: -1, 0, 1]) is 1
+
+  math.min([list: -1, -10, 1/2]) is -10
+  math.min([list: 1/10, 1/100, 1/100000000000000000000000]) is 1/100000000000000000000000
+
+	math.arg-min([list:]) raises "empty"
```
Comment (sorawee) [inline tests/pyret/tests/test-math.arr]: inconsistent indentation.

File: tests/pyret/tests/test-statistics.arr (hunk 1 above)
Comment (SDooman) [inline tests/pyret/tests/test-statistics.arr]: yikes!  thanks for catching this

**Comment by jpolitz** [comment]
I cleaned up some uses of list imports to make this work with both `provide *` and `include`

**Pull request merged**